homeassistant/components/mqtt/vacuum/schema_legacy.py

```diff
@@ -236,66 +272,64 @@ def _setup_from_config(self, config):
                 CONF_ERROR_TEMPLATE,
                 CONF_FAN_SPEED_TEMPLATE,
             )
+            if key in config
         }
 
-    def _prepare_subscribe_topics(self):
+    def _prepare_subscribe_topics(self) -> None:
         """(Re)Subscribe to topics."""
-        for tpl in self._templates.values():
-            if tpl is not None:
-                tpl = MqttValueTemplate(tpl, entity=self)
 
         @callback
         @log_messages(self.hass, self.entity_id)
-        def message_received(msg):
+        def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT message."""
             if (
                 msg.topic == self._state_topics[CONF_BATTERY_LEVEL_TOPIC]
-                and self._templates[CONF_BATTERY_LEVEL_TEMPLATE]
+                and CONF_BATTERY_LEVEL_TEMPLATE in self._config
             ):
-                battery_level = self._templates[
+                battery_level: ReceivePayloadType = self._templates[
                     CONF_BATTERY_LEVEL_TEMPLATE
-                ].async_render_with_possible_json_value(msg.payload, None)
+                ](msg.payload, "")
```

Comment:
Given how many times `self._templates[CONF_X](msg.payload, "")` repeats in this function, maybe it should be refactored to `self._render_template(conf_key, msg, default="")`?

Reply:
Feel free to open a PR